src/dep.cc

```diff
@@ -660,6 +660,8 @@ class DepBuilder {
     if (!PickRule(output, n, &rule_merger, &pattern_rule, &vars)) {
       return n;
     }
+
+    // TODO(lberki): When can rule_merger be null?
```

Comment:
Hey, I didn't realize we don't have those! It's at least a bit more sanity in those Makefiles. I'll remove this comment in any case, since I don't want to rock the boat with this change. Do you think it would make sense to also remove the implementation of implicit / suffix rules from Kati then so that the code can be that much simpler?

Reply:
Yep, I added the flags in #136 and we've had them turned for a while now. We're generally putting all of these types of changes behind flags, since we know people are using kati for other purposes than just Android.

One of the things I was looking at last week was how much simpler(/faster) we could make Kati now that we've removed a lot of complexities of Make. Certain areas could be rearchitected in a much cleaner way, but I'm still not entirely sure it's worth the time to do anything more than tactical improvements, at least until a few more things are removed (and at that point there's significantly diminishing returns, as big changes have compatibility periods that encroach on the timeline where we hope to remove the vast majority of make code in Android)